ADR/ADR005-use-multi-tenant-runner.md

```diff
@@ -0,0 +1,83 @@
+# ADR005: Use multi-tenant runner
+
+Date: 2022-04-07
+
+## Status
+
+Accepted
+
+## Context
+
+The forms platform has a "runner", the service that runs the forms itself to the user who is filling them in.
+
+There are two different ways to approach the runner - single or multi tenant. Two existing forms creation services (MOJ Forms, XGovFormBuilder) both use the single tenant approach.
+
+### Single tenant structure
+
+```mermaid
+flowchart TD
+  Manager[Form Manager]
+  A[Form A]
+  B[Form B]
+  C[Form C]
+  D[Form D]
+
+  subgraph RunnerA[Runner]
+    A
+  end
+
+  subgraph RunnerB[Runner]
+    B
+  end
+
+  subgraph RunnerC[Runner]
+    C
+  end
+
+  subgraph RunnerD[Runner]
+    D
+  end
+
+  Manager --> RunnerA
+  Manager --> RunnerB
+  Manager --> RunnerC
+  Manager --> RunnerD
+```
+
+In this structure, each form has its own runner which serves each form to the end user. This:
+
+- Limit the impact if one form runner encounters an error
+- Makes it more complicated to roll out updates to each runner as we scale
+- Requires us to run a server consistantly for all forms regardless of traffic, using up money + energy when not required
+
+### Multi tenant structure
+
+```mermaid
+flowchart LR
+  Manager[FormManager]
+  subgraph Runner
+    A[Form A]
+    B[Form B]
+    C[Form C]
+    D[Form D]
+  end
+  Manager --> Runner
+```
+
+In this structure, there is one runner service that serves forms to the end user. This:
+
+- Enables us to have easier control over updating the runner where required (dependency version increases, service updates, etc.)
+- Enables the ability to scale to the overall load on the forms service, removing the need to run idle servers for forms that have no traffic
+- Makes errors in the forms runner service have wider impact as all forms will be effected rather than just the one on the runner encountering the error
```

Comment:
Increases the risk of a problem with one form affecting the performance of others